src/main/java/bio/terra/service/duos/DuosService.java

```diff
@@ -148,13 +152,97 @@ private void updateLastSynced(DuosFirecloudGroupModel firecloudGroup, Instant la
     try {
       duosDao.updateFirecloudGroupLastSyncedDate(id, lastSyncedDate);
     } catch (CannotSerializeTransactionException ex) {
-      String msg =
-          "Members of Firecloud group %s were updated, "
-              + "but could not update DUOS Firecloud Group record %s with last_synced_date %s"
-                  .formatted(firecloudGroup.getFirecloudGroupName(), id, lastSyncedDate);
+      String message =
+          firecloudGroup.getFirecloudGroupEmail()
+              + " members were updated, "
+              + "but an error occurred when updating its database record with last_synced_date "
+              + lastSyncedDate;
       String errorDetail =
           "This indicates that another process was updating the database row at the same time";
-      throw new DuosFirecloudGroupUpdateConflictException(msg, ex, List.of(errorDetail));
+      throw new DuosFirecloudGroupUpdateConflictException(message, ex, List.of(errorDetail));
+    }
+  }
+
+  /**
+   * For all DUOS dataset IDs registered in TDR, force a sync of their Firecloud group members. A
+   * failure to sync any Firecloud group(s) should not block attempts to sync the rest.
+   *
+   * @return DUOS Firecloud groups whose members were successfully synced, and any errors occurred
+   *     which may have interfered with syncing
+   */
+  public DuosFirecloudGroupsSyncResponse syncDuosDatasetsAuthorizedUsers() {
+    List<ErrorModel> errors = new ArrayList<>();
+
+    // 1. Parallelize our calls to external systems (DUOS, Sam) which perform the sync
+    Instant lastSyncedDate = Instant.now();
+    List<Future<UUID>> futureSyncedIds =
+        duosDao.retrieveFirecloudGroups().stream()
+            .map(group -> executor.submit(() -> syncFirecloudGroupContents(group, errors)))
+            .toList();
+    List<UUID> syncedIds =
+        FutureUtils.waitFor(futureSyncedIds).stream().filter(Objects::nonNull).toList();
```

Comment:
Using the JDK parallel stream APIs avoid the need to work with `Future`s directly.
```suggestion
    List<UUID> syncedIds =
        duosDao.retrieveFirecloudGroups().parallelStream()
            .map(group -> syncFirecloudGroupContents(group, errors)))
            .filter(Objects::nonNull)
            .toList();
```

Or, if you want to keep using `executor`, this should work:

```suggestion
    var groups = duosDao.retrieveFirecloudGroups();
    List<UUID> syncedIds =
       executor.submit(() ->
        groups.parallelStream()
            .map(group -> syncFirecloudGroupContents(group, errors)))
            .filter(Objects::nonNull)
            .toList());
```

Reply:
Neat!  This worked well on my dev deployment  (8 FC groups synced in <2 seconds).  The scale of this integration is unlikely to be dramatically larger, so I think sidestepping the performance threadpool is reasonable here.

Reply:
Actually, after talking about this in our team's stand-up I'm going to move back to the original FutureUtils method for consistency within our codebase.  CC: @nmalfroy